### ISSUE TITLE
Fix the issued warning when the demominator is zero in the "colour.XYZ_to_xyY" definition.

### ISSUE THREAD

**Issue by starkfan007** [colour-science/colour#907]
In the XYZ_to_xyY conversion function, if X+Y+Z is equal to 0, the following warning will be generated: 'invalid value encountered in true_divide'，we can use `np.finfo(np.float32).eps` to replace possible zeros.
```
    eps = np.finfo(np.float32).eps
    xyY = np.where(
        np.all(XYZ == 0, axis=-1)[..., np.newaxis],
        XYZ_n,
        tstack([
            X / np.maximum(X + Y + Z, eps),
            Y / np.maximum(X + Y + Z, eps),
            from_range_1(Y),
        ]),
    )
```

**Comment by KelSolaar** [comment]
Hi @starkfan007,

Sorry for the late reply! Yes, we are aware of that, purposely, the intent has always been for the user to deal with NaN and Inf themselves. For example, it could be argued that 0 luminance should never occur and hiding that from the user could generate unexpected behaviour downstream where a warning clearly and explicitly hints at something going on that should be looked at.

With that said, and as I'm going through the entire codebase as I'm adding typing annotations, I noticed a few places where we are doing things inconsistently. I think that similarly to [colour.algebra.spow](https://colour.readthedocs.io/en/develop/generated/colour.algebra.spow.html#colour.algebra.spow), we will add a `sdiv` that does a similar job.

**Comment by KelSolaar** [comment]
@starkfan007 this should have been addressed in #955!
